tools/omero/omero_import.xml

```diff
@@ -0,0 +1,125 @@
+    <description></description>
+    <macros>
+        <token name="@TOOL_VERSION@">5.18.0</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+    </macros>
+    <xrefs>
+        <xref type="bio.tools">omero</xref>
+    </xrefs>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">omero-py</requirement>
+        <requirement type="package" version="21.0.2">openjdk</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        mkdir folder &&
+        #for $image in $folder:
+            ln -s '$image' folder/'$image.element_identifier'.$image.ext &&
+        #end for
+
+        omero import folder
+
+        #if $predefined_cond.predefined_sel == 'yes'
+            -s $predefined_cond.omero_host.fields.server
+            -p $predefined_cond.omero_host.fields.port
+            -u $__user__.extra_preferences.get('omero_'+$predefined_cond.omero_host.fields.value+'|username', $test_username)
+        #else
+            -s $predefined_cond.omero_host
+            -p $predefined_cond.omero_port
+            -u $__user__.extra_preferences.get('omero|username', $test_username)
+        #end if
+        > $log
+    ]]></command>
+    <environment_variables>
+        <environment_variable name="OMERO_PASSWORD"><![CDATA[#if $predefined_cond.predefined_sel == 'yes':
+$__user__.extra_preferences.get('omero_'+$predefined_cond.omero_host.fields.value+'|password', $test_password)
+#else
+$__user__.extra_preferences.get('omero|password', $test_password)
+#end if]]></environment_variable>
+    </environment_variables> 
+    <inputs>
+        <param name="folder" type="data" format="bmp" multiple="true" optional="false" label="Images"/>
+        <conditional name="predefined_cond">
+            <param name="predefined_sel" type="select" label="Use a predfinced OMERO server">
+                <option value="yes">Yes</option>
+                <option value="no">No</option>
+            </param>
+            <when value="yes">
+                <param name="omero_host" type="select" label="OMERO server">
+                    <options from_data_table="omero_servers"/>
+                    <validator type="no_options" message="No OMERO servers are defined. Contact your Galaxy administrator."/>
+                </param>
+            </when>
+            <when value="no">
+                <param name="omero_host" type="text" label="OMERO host URL">
+                    <validator type="regex" message="Enter a valid host location, for example, your.omero.server">^[a-zA-Z0-9._-]*$</validator>
+                </param>
+                <param name="omero_port" type="integer" value="4064" optional="false" label="OMERO port"/>
+            </when>
+        </conditional>
+        <param name="test_username" type="hidden" value=""/>
+        <param name="test_password" type="hidden" value=""/>
```

Comment:
If this is the way you want to make it testable then that's fine of course, but I think we should be able to set these in the test definition instead.

Reply:
> How would you do this?

<test>
    <user_preferences>
        <param name="test_username">user123</param>
    </user_preferences>
</test>

and then set these up as you run the test.

Reply:
Cool idea. Will open a feature request in the Galaxy repo.

Reply:
(Again, this is fine if you want to make progress on this, this isn't implemented now)